src/Lean/Language/Basic.lean

```diff
@@ -135,32 +135,36 @@ class ToSnapshotTree (α : Type) where
   toSnapshotTree : α → SnapshotTree
 export ToSnapshotTree (toSnapshotTree)
 
+/-- Runs a tree of snapshots to conclusion, incrementally performing `f` on each snapshot in tree preorder. -/
+@[specialize] partial def SnapshotTree.forM [Monad m] (s : SnapshotTree)
+    (f : Snapshot → m PUnit) : m PUnit := do
+  match s with | mk element children => f element; children.forM (·.get.forM f)
+
```

Comment:
@Kha I added this method to make `runAndReport` and `getAll` one-liners and to avoid a potential linearity issue with the projections. As this is stylistic refactor, I wanted to highlight it to you (in case you do not approve).

Reply:
The function great, just the formatting of the body is unusual. I would definitely write this as four lines, not one.